swap/src/asb/tracing.rs

```diff
@@ -1,18 +1,28 @@
+use std::path::Path;
+
 use anyhow::Result;
 use tracing_subscriber::filter::LevelFilter;
 use tracing_subscriber::fmt::time::UtcTime;
+use tracing_subscriber::fmt::writer::MakeWriterExt;
 use tracing_subscriber::FmtSubscriber;
 
-pub fn init(level: LevelFilter, json_format: bool, timestamp: bool) -> Result<()> {
+pub fn init(
+    level: LevelFilter,
+    json_format: bool,
+    timestamp: bool,
+    dir: impl AsRef<Path>,
+) -> Result<()> {
     if level == LevelFilter::OFF {
         return Ok(());
     }
 
     let is_terminal = atty::is(atty::Stream::Stderr);
 
+    let appender = tracing_appender::rolling::never(dir.as_ref(), "swap-all.log");
+
     let builder = FmtSubscriber::builder()
         .with_env_filter(format!("asb={},swap={}", level, level))
-        .with_writer(std::io::stderr)
+        .with_writer(appender.and(std::io::stderr))
```

Comment:
We should make sure the logs always get written to disk in JSON in the same format regardless of whether the --debug or --json flag is enabled.

This already is the case for the cli. I don't know if this also applies here.